sklearn/linear_model/_base.py

```diff
@@ -662,8 +662,10 @@ def fit(self, X, y, sample_weight=None):
             X, y, accept_sparse=accept_sparse, y_numeric=True, multi_output=True
         )
 
-        if sample_weight is not None:
-            sample_weight = _check_sample_weight(sample_weight, X, dtype=X.dtype)
+        if sample_weight is not None:  # the weights should be non-negative
```

Comment:
There is no need for this comment since passing the keyword argument is self-explanatory.

```suggestion
        if sample_weight is not None:
```